### ISSUE TITLE
Updating multivariate weighting in one environment appears to update in all

### ISSUE THREAD

**Issue by matthewelwell** [SolidStateGroup/bullet-train-api#421]
The issue seems to stem from confusion around the difference between the`default_percentage_allocation` value and the 
`percentage_allocation` value. The `default_percentage_allocation` value should only be set once, on creation, similar 
to the `default_enabled` value of a feature state.

From what I can tell, the FE is using the `default_percentage_allocation` value to display the current percentage 
allocation for a variant in a given environment which is incorrect. It is also updating this value whenever the user
updates the value in a given environment. 

Here is an example journey with some of these key points highlighted: 

### Step 1. User creates a new feature with a multivariate option. 

POST request made to `/projects/:id/features/` with the following request body:

```json
{
  "name": "mvtest2",
  "initial_value": "control",
  "default_enabled": true,
  "multivariate_options": [
    {
      "type": "unicode",
      "boolean_value": null,
      "integer_value": null,
      "string_value": "variant-1",
      "default_percentage_allocation": 5
    }
  ],
  "project": "3053",
  "type": "MULTIVARIATE"
}
```

### Step 2. User reviews features in an environment

GET request made to `/api/v1/features/featurestates/?environment=:id&feature=:id` with the following response body

```json
{
  "count": 1,
  "next": null,
  "previous": null,
  "results": [
    {
      "id": 70904,
      "feature_state_value": {
        "type": "unicode",
        "string_value": "control",
        "integer_value": null,
        "boolean_value": null
      },
      "multivariate_feature_state_values": [
        {
          "id": 2352,
          "multivariate_feature_option": 686,
          "percentage_allocation": 5.0
        }
      ],
      "enabled": true,
      "feature": 13309,
      "environment": 7478,
      "identity": null,
      "feature_segment": null
    }
  ]
}
```

Note the value of `results.0.multivariate_feature_state_values.0.percentage_allocation` is 5.0 as expected. As I 
understand it, however, this is not the value that the FE is using to display to the user, it's using the value of
`default_percentage_allocation` that it retrieves using the response from `/api/v1/projects/:id/features/`. 

### Step 3. User updates the percentage weighting in a given environment

At this point, 2 PUT requests are made 

One to `/api/v1/environments/:id/featurestates/:id/` with the following request body

```json
{
  "id": 70903,
  "feature_state_value": "control",
  "multivariate_feature_state_values": [
    {
      "id": 2351,
      "multivariate_feature_option": 686,
      "percentage_allocation": 10
    }
  ],
  "identity": null,
  "enabled": true,
  "feature": 13309,
  "environment": 7477,
  "feature_segment": null
}
```

This is correct. 

The FE also makes another PUT request to `/api/v1/projects/:id/features/:id/` with the following body: 

```json
{
  "id": 13309,
  "name": "mvtest2",
  "type": "MULTIVARIATE",
  "default_enabled": true,
  "initial_value": "control",
  "created_date": "2021-10-19T10:10:21.307968Z",
  "description": null,
  "tags": [],
  "multivariate_options": [
    {
      "id": 686,
      "type": "unicode",
      "integer_value": null,
      "string_value": "variant-1",
      "boolean_value": null,
      "default_percentage_allocation": 10
    }
  ],
  "is_archived": false,
  "project": "3053"
}
```

The `default_percentage_allocation` value on the multivariate options should not be updated in this request, this value 
should only be set when creating a multivariate option. 

So, in summary, 2 issues: 

 1. FE is displaying the `default_percentage_allocation` instead of `percentage_allocation`. 
 2. FE is updating the `default_percentage_allocation` value of already existing multivariate options.

**Comment by kyle-ssg** [comment]
https://github.com/Flagsmith/flagsmith/pull/423
